### PR TITLE
Ability to run tests against remote host

### PR DESCRIPTION
**Pull request by dacrome**

### CODE REVIEW

**Comment by tkrille** [comment]
Is this final now?

**Comment by dacrome** [comment]
Yes

File: pom.xml [hunk 1]
```diff
@@ -600,6 +608,7 @@
             <id>mysql</id>
             <properties>
                 <database.container>mysql</database.container>
+                <osiam.test.host>${osiam.host.protocol}://${osiam.host}</osiam.test.host>
```
Comment (tkrille) [inline pom.xml]: I don't see why we need this property in 2 different profiles that always have the same value regardless of the active profile.

Comment (dacrome) [inline pom.xml]: Absolutely right, I removed both and added the property to the `properties` element

**Comment by tkrille** [comment]
Is this enabling me to run the ITs against the Docker dev image?

**Comment by dacrome** [comment]
> Is this enabling me to run the ITs against the Docker dev image?

Yes, but then we also need to make the [osiam tomcat port configurable](https://github.com/osiam/connector4java-integration-tests/pull/278/files#diff-bcaea6e7a59c9927f3d4e6be6b341988R72) instead of using the fixed `8180` port. Should I change this? I think about the property `osiam.host.tomcat.port`.

If this is configurable you can run `mvn failsafe:integration-test -Dosiam.host=osiamhost -Dosiam.host.tomcat.port=8080` from command line. Run from IDE only when configuring it before.

**Comment by tkrille** [comment]
> I think about the property `osiam.host.tomcat.port`.

Why not just a plain `osiam.port`?

**Comment by tkrille** [comment]
How am I supposed to tell my IDE about the endpoints? Do I have to set it on the run configuration or change the sources of the ITs?

**Comment by dacrome** [comment]
> Why not just a plain osiam.port?

Yes, that's better. But I think about to pass the host and port in the same property `osiam.host` e.g. `localhost:8180`. What do you think?

> How am I supposed to tell my IDE about the endpoints? Do I have to set it on the run configuration or change the sources of the ITs?

Yes, you need to set the system properties in the run configuration or you change the properties in the [`pom.xml`](https://github.com/osiam/connector4java-integration-tests/pull/278/files#diff-bcaea6e7a59c9927f3d4e6be6b341988R72). But first we need to agree how to set the osiam port.

**Comment by tkrille** [comment]
Do you really need the port being configurable?

**Comment by dacrome** [comment]
> Do you really need the port being configurable?

Yes, then it's possible to run the ITs against every remote osiam host. For sure the docker port must be exposed. And the database port must also be configurable e.g. with the property `osiam.database.port`.

Then the following properties can be configured:
- `docker.host`
- `osiam.host`
- `osiam.host.protocol`
- `osiam.port`
- `osiam.database.port`

To run them against a docker host e.g. in a boot2docker vm, you only need to set the first two properties, the rest is set with the default values of the integration-tests.

**Comment by tkrille** [comment]
What about the mail server port?

**Comment by dacrome** [comment]
> What about the mail server port?

Yes, `osiam.mail.host` and `osiam.mail.port` should also be an property to configure the mail server access.

**Comment by tkrille** [comment]
Ok, just tell me when you're ready.

**Comment by dacrome** [comment]
Ready!

File: README.md [hunk 2]
```diff
@@ -85,3 +85,32 @@ Just use the normal remote debugging setup for your IDE and connect to `localhos
 Set your breakpoints as usual and run the test in the ITs project.
 Your IDE should pop up as soon as the service reaches the breakpoint.
 
+## Run against remote docker host
+
+If you like to run the tests against a remote docker host, you nedd to set the
+following system properties:
+
+Docker:
```
Comment (tkrille) [inline README.md]: What about using a heading here?

File: README.md [hunk 3]
```diff
@@ -85,3 +85,32 @@ Just use the normal remote debugging setup for your IDE and connect to `localhos
...
+- `docker.host`
+  The URL of the docker daemon. Default: `http://localhost:2375`
+
+OSIAM:
```
Comment (tkrille) [inline README.md]: What about using a heading here?

File: README.md [hunk 4]
```diff
@@ -85,3 +85,32 @@ Just use the normal remote debugging setup for your IDE and connect to `localhos
...
+- `osiam.host.protocol`
+  The protocol of the OSIAM host. Default: `http`
+- `osiam.host`
+  The host where OSIAM is running. Default: `localhost`
+- `osiam.port`
+  The port where OSIAM is running. Default: `8180`
+- `osiam.database.host`
+  The host where the database for OSIAM is running. Default: `localhost`
+- `osiam.database.port`
```
Comment (tkrille) [inline README.md]: I cannot find this property in the pom, i.e. it's default value is not set.

Comment (tkrille) [inline README.md]: Oh, I see. Never mind.

**Comment by tkrille** [comment]
I can't figure out how to run these against the Docker image. Maven will always start the containers locally and then connect to my running dev container. Am I right, if I assume that this is mostly for Mac/Windows users, who cannot run Docker natively, but run the ITs against a VM with Docker pre-installed, etc.?

**Comment by dacrome** [comment]
Yes. But you can test this also on linux by [install docker machine](https://docs.docker.com/machine/install-machine/#on-linux).

File: README.md (hunk 3 above)
Comment (dacrome) [inline README.md]: Sounds about right

File: README.md (hunk 2 above)
Comment (dacrome) [inline README.md]: Sounds about right

**Pull request merged**